src/Handlers/MainContainer.cpp

```diff
@@ -19,7 +22,8 @@ int MainContainer::Init () {
 			c.append_attribute("cross").set_value("GPU");
 		#endif
 		solver->configfile.save_file(filename);
```

Comment:
You cannot write the xml file after the execution, because the execution can fail, or can be stopped - and you would still want the xml file to be created.